### DISCUSSION TITLE
"no request found" when reading config data in init_by_lua_block

### DISCUSSION THREAD

**Discussion by jackshang** [apache/apisix#6015] [General]
I want to read initialize config data from ETCD in init_by_lua_block, but something went wrong. I found Apisix does the same way. Can anyone tell me the reason? thx

2022/01/05 10:14:46 [error] 7000#13240: init_by_lua error: ./lualib/resty\http.lua:133: no request found
stack traceback:
	[C]: in function 'ngx_socket_tcp'
	./lualib/resty\http.lua:133: in function 'new'
	./lualib/resty\etcd\v3.lua:133: in function '_request_uri'
	./lualib/resty\etcd\v3.lua:332: in function 'refresh_jwt_token'
	./lualib/resty\etcd\v3.lua:114: in function '_request_uri'
	./lualib/resty\etcd\v3.lua:510: in function 'readdir'

**Comment by starsz** [comment]
Hi @jackshang Since the cosocket doesn't support at the `init_by_lua_block`.
Refer: https://github.com/openresty/lua-nginx-module#ngxsockettcp

Maybe you can use lua socket instead or do `require("apisix.patch").patch()` before your code.
